cpp/cgi_progs/zotero_cgi.cc

```diff
@@ -570,17 +570,22 @@ bool ProcessShowQASubActionAdd(const std::multimap<std::string, std::string> &cg
     // sub-action "add"
     const std::string add_type(GetCGIParameterOrDefault(cgi_args, "add_type"));
     const std::string add_tag(GetCGIParameterOrDefault(cgi_args, "add_tag"));
+    const std::string add_subfield_code(GetCGIParameterOrDefault(cgi_args, "add_subfield_code"));
+    const std::string add_regex(GetCGIParameterOrDefault(cgi_args, "add_regex"));
     const std::string add_presence(GetCGIParameterOrDefault(cgi_args, "add_presence"));
     if (add_type.empty() or add_tag.empty() or add_presence.empty())
         return false;
 
-    std::string journal_id_to_insert = db_connection->escapeAndQuoteString(zeder_journal_id);
-    if (add_type == "global")
-        journal_id_to_insert = "NULL";
+    const std::string regex_to_insert(add_regex.empty() ? "NULL"
+                                                        : db_connection->escapeAndQuoteString(add_regex));
+    const std::string journal_id_to_insert(add_type == "global" ? "NULL"
+                                                                : db_connection->escapeAndQuoteString(zeder_journal_id));
```

Comment:
the variable should  be renamed to journal_id as well

Reply:
Done